src/statemachine.rs

```diff
@@ -584,6 +584,8 @@ impl StateMachine {
                         }
                     };
 
+                    send_status(&status, crate::StatusUpdate::PresenceRequired);
```

Comment:
I feel like the intent is more clear if this is closer to where we actually send the make credential message to the device?

Reply:
That's fair. I moved the status updates to immediately before sending the command in each case.